### ISSUE TITLE
Vite: Application not prioritizing or reading paths from project's tsconfig.json

### ISSUE THREAD

**Issue by NordlingDev** [nxext/nx-extensions#702]
**Describe the bug**
Vite application does not look for any paths configuration inside the project's `tsconfig.json`. It only looks for the paths under workspace's root `tsconfig.base.json`. This is not how paths are resolved in official @nrwl apps and it prevents us from overriding paths with custom paths per project.

**To Reproduce**
1. Generate a simple Vite application: `nx g @nxext/vite:application my-app`
2. Add paths config to application's `tsconfig.json`:
```
...
"compileOptions": {
  ...
  "paths": {
    "~/*": ["apps/my-app/src/app/*"]
  }
}
```
3. Following error occurs when serving the app: `[vite] Internal server error: Failed to resolve import "~/foo" from "apps\my-app\src\app\app.ts". Does the file exist?`

**Expected behavior**
As official nx apps work, we should be able to override the `tsconfig.base.json`'s paths by redefining it under the application's own `tsconfig.json`.

**Comment by merlosy** [comment]
Hi! This might not be of great help, but there used to be a webpack plugin to do this: [tsconfig-paths-webpack-plugin](https://www.npmjs.com/package/tsconfig-paths-webpack-plugin).
As of nx^13, it is part of several nx packages' dependencies.

**Comment by NordlingDev** [comment]
@merlosy Yes but it seems like this plugin completely overrides that config.
